packages/scripts/src/findFaq.ts

```diff
@@ -166,3 +174,127 @@ export const findFaq = async ({
 
   return faqEntries;
 };
+
+/**
+  Make a wrapper around the given collection that conforms to the VectorStore
+  interface.
+
+  Does not manage the collection - callers are still responsible for closing the
+  client.
+ */
+export const makeFaqVectorStoreCollectionWrapper = (
+  collection: Collection<WithId<FaqEntry & { created: Date; epsilon: number }>>
+): VectorStore<WithId<FaqEntry>> => {
+  return {
+    findNearestNeighbors(vector, options) {
+      const {
+        indexName,
+        path,
+        k,
+        minScore,
+        filter,
+        numCandidates,
+      }: Partial<FindNearestNeighborsOptions> = {
+        // Default options
+        indexName: "vector_index",
+        path: "embedding",
+        k: 10,
+        minScore: 0.95,
+        // User options override
+        ...(options ?? {}),
+      };
+      return collection
+        .aggregate<WithScore<WithId<FaqEntry>>>([
+          {
+            $vectorSearch: {
+              index: indexName,
+              queryVector: vector,
+              path,
+              limit: k,
+              numCandidates: numCandidates ?? k * 15,
+              filter,
+            },
+          },
+          {
+            $addFields: {
+              score: {
+                $meta: "vectorSearchScore",
+              },
+            },
+          },
+          {
+            $match: {
+              score: { $gte: minScore },
+            },
+          },
+        ])
+        .toArray();
+    },
+  };
+};
+
+/**
+  For each given question, finds if any similar messages already have a faqId.
+  If so, adopts the faqId. Otherwise, invents a new faqId for this category of
+  question.
+ */
+export const assignFaqIds = async ({
+  faq,
```

Comment:
```suggestion
  faqs,
```

```suggestion
  faqEntries,
```

Thoughts? I think it's useful to have `faq` => one cluster of questions and `faqs` => `faq[]`.

Would entail some small updates throughout